samples/GSC/scene_shared.csc

```diff
@@ -0,0 +1,2390 @@
+#using scripts\codescripts\struct;

```

Comment:
Wow!! This file is huge and seems to nearly repeat similar segments of code. Are you able to replace this with a smaller representative sample with less apparent duplication?